WordPress/Classes/ViewRelated/Domains/Register/RegisterDomainSuggestionsViewController.swift

```diff
@@ -0,0 +1,113 @@
+import UIKit
+import WordPressAuthenticator
+
+class RegisterDomainSuggestionsViewController: NUXViewController, HidableBottomViewOwner {
+
+    @IBOutlet weak var bottomViewBottomConstraint: NSLayoutConstraint!
+    @IBOutlet weak var bottomViewHeightConstraint: NSLayoutConstraint!
+    private var domain: String?
+
+    private var domainsTableViewController: SiteCreationDomainsTableViewController?
```

Comment:
I'm still not sure about this approach. It's just confusing having `domainsTableViewController` being of type `SiteCreationDomainsTableViewController` as the Register Domain flow has nothing to do with Site Creation. Granted, the functionality is quite similar. And I see that you're trying to reuse code, which is great. But we need a clear structure that is not so specific.

Maybe a good example is the `NUXTableViewController`. It contains common functionality while still allowing some properties and methods to be overridden. It's not specific to any one feature, and is obvious that it's meant to be used generically. Then other VCs, like `SiteCreationDomainsTableViewController` that is specific to a feature, use that base class for its functionality and override where necessary.

That may not not be approach you decide to take. However, essentially what we try to do is segregate specific functionality from generic functionality. `SiteCreationDomainsTableViewController` is specific. If we want to reuse its functionality, it should be more generic.

Reply:
I'll do this next, I just didn't have the chance

Reply:
Done. I've extracted a base class. I'll be sending the new commits today